sec1_intro_and_setup/test_app/config/initializers/content_security_policy.rb

```diff
@@ -0,0 +1,30 @@
+# Be sure to restart your server when you modify this file.
```

Comment:
Whats this file for?

Reply:
This file controls the data that is allowed to flow into your app and from what external sources.

Reply:
How does it control data that flow into your app from external sources?

Reply:
1. Using CSP (control security policy), the CSP is like a whitelist system where the developers or website operator use to tell the client which external resources can be loaded and executed. Those that are not on the list are not allowed to flow into the app.
2. Resources that is linked to the app, for example, images, linking scripts, and etc.